internal/handlers/common/projection.go

```diff
@@ -236,6 +236,8 @@ func ProjectDocument(doc, projection, filter *types.Document, inclusion bool) (*
 		return nil, err
 	}
 
+	projected.SetRecordID(doc.RecordID())
```

Comment:
Any idea how it can be done better? Currently `find` applies sequences of iterator and project iterator creates a new document and didn't copy recordID.